Integrate gutenberg-mobile release 1.58.0

SiobhyB opened the pull request:
## Description
This PR incorporates the 1.58.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3746

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Release script: Update gutenberg-mobile ref

SiobhyB committed:
Update RELEASE-NOTES

SiobhyB committed:
Podfile updates

SiobhyB committed:
Podfile updates

fluiddot commented on RELEASE-NOTES.txt:
```diff
@@ -2,6 +2,7 @@
 -----
 * [internal] Converted part of Comment model to Swift. Should be no functional changes. [#16898]
 * [*] Enables Support for Global Style Colors with Full Site Editing Themes [#16823]
+* [***] New Block: Embed block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3727]
```
Not sure if it's standard but I noticed that other entries related to changes in the editor include `Block editor: ` message at the beginning, what do you think about including it too?

SiobhyB replied:
Done in https://github.com/wordpress-mobile/WordPress-iOS/pull/16911/commits/6c10ffea92c45529f6de51580e1efcca3bc8ba00. 🙇‍♀️

SiobhyB committed:
Add "Block editor" indicator to RELEASE-NOTES

This brings the notes in line with previous releases, which have included a "Block editor" indicator with any changes coming from Mobile Gutenberg.

SiobhyB committed:
Podfile updates

fluiddot reviewed (approved):
LGTM 🎊 !

I did a quick test (build `53488`) with iPhone 11 (iOS 14.2) and everything worked fine.

**NOTE:** Before merging this PR, we have to change the Gutenberg Mobile ref to point to the release tag `v1.58.0`.

SiobhyB committed:
Podfile updates

SiobhyB committed:
Merge branch 'develop' into gutenberg/integrate_release_1.58.0

SiobhyB committed:
Checksum update
